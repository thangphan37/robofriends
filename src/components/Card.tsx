import React from 'react';
import {Robot} from './CardList';

const Card: React.FC<{robot: Robot}> = ({robot}) => {
  const { id, username, email } = robot;
  return(
    <div className="pa3 grow bg-light-green ma2 tc bw2 br3 shadow-5 dib">
      <img src={`https://robohash.org/${id}?size=200x200`} alt="robot"/>
      <h2>{username}</h2>
      <p>{email}</p>
    </div>
  )
}

export default Card;