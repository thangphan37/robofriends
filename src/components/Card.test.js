import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';


it('snapshot Card component', () => {
  const mockRobot = {
    id: 1,
    username: 'john',
    email: 'john@gmail.com'
  };

  const wrapper = renderer.create(<Card robot={mockRobot}/>).toJSON();

  expect(wrapper).toMatchSnapshot();
})
