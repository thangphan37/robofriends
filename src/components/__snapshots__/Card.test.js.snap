// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Card component 1`] = `
<div
  className="pa3 grow bg-light-green ma2 tc bw2 br3 shadow-5 dib"
>
  <img
    alt="robot"
    src="https://robohash.org/1?size=200x200"
  />
  <h2>
    john
  </h2>
  <p>
    john@gmail.com
  </p>
</div>
`;
